pgroll-integration.mdx

```diff
@@ -0,0 +1,87 @@
+
+With our public beta today, pgroll is now the underlying technology used behind our schema change APIs for the Xata [web application](http://app.xata.io), CLI and SDK. Using pgroll behind the scenes allows us to dogfood our product, as well as de-risk migrations by providing a standardized API. It also means that users will have more find grained control over their databases.
+
+## New features
+
+Users have consistently voiced a number of pain points, from adding constraints, to column editing, to having more flexible schemas during the initial data modelling phases. Using the capability pgroll provides behind our direct access to postgres feature allows us to address a lot of these issues.
+
+Users will notice some new features such as:
+
+- **The ability to add and remove constraints after creating columns**
+  - For example, modifying a column to be not null or unique after a column has been created and contains data is now possible.
+- **The ability to create columns that are not nullable with no defaults**
+  - These columns will be populated with a “zero” column value depending on the column’s type.
+- **The ability to run raw arbitrary SQL migrations on the underlying database to create column types outside of the known Xata types, as well as constraints, foreign keys, and other Postgres native supported objects.**
+
+## Enabling direct Postgres access
+
+In order to access the new features users must first do two things.
+
+    1. Enable direct access for your workspace
+
+    ![Enable direct access for your workspace](/images/pgroll-integration/workspace.png)
+
+    2. Enable direct access when creating their database
+
+        ![Enable direct access for your database](/images/pgroll-integration/database.png)
+
+## New migration experience
+
+Once done, there will be new options to edit columns on the Table view and Schema view.
+
+![New edit column functionality](/images/pgroll-integration/edit.png)
+
+There will also be a new tab called Migration Editor on the Schema view page. Users can run arbitrary migrations with type safe completions. A set of examples can be found in the [pgroll repository](https://github.com/xataio/pgroll/tree/main/examples).
+
+![New Migration Editor](/images/pgroll-integration/editor.png)
+
+The Schema History tab has also had makeover. The Schema History tab contains a list of all the migration operations that have been run a the user’s database schema up until now. Information such as the migrations status, migration type (whether it was run using pgroll or directly with SQL), and contents can also be viewed.
+
+![New Migration History tab](/images/pgroll-integration/history.png)
+
+Note that our internal columns have been renamed. We now use `xata_id` rather than `id` as the table's primary key, and `xata_updatedat`, `xata_createdat`, and `xata_version` for internal metadata.
```

Comment:
```suggestion
For existing Xata users, note that our internal columns have been renamed. We now use `xata_id` rather than `id` as the table's primary key, and `xata_updatedat`, `xata_createdat`, and `xata_version` for internal metadata.
```

Reply:
Removed this section altogether since it was repeated below.